src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala

```diff
@@ -14,17 +19,92 @@ trait RawlsBillingProjectComponent {
-    rawlsBillingProjectQuery insertOrUpdate billingProject map { _ => billingProject }
+    def user = foreignKey("FK_PROJECT_USERS_USER", userSubjectId, rawlsUserQuery)(_.userSubjectId)
+    def project = foreignKey("FK_PROJECT_USERS_PROJECT", projectName, rawlsBillingProjectQuery)(_.projectName)
+    def pk = primaryKey("PK_PROJECT_USERS", (userSubjectId, projectName))
   }
 
-  def loadBillingProject(name: String) = rawlsBillingProjectByNameQuery(name).result
+  protected val projectUsersQuery = TableQuery[ProjectUsersTable]
+  private type RawlsBillingProjectQuery = Query[RawlsBillingProjectTable, RawlsBillingProjectRecord, Seq]
+  private type ProjectUsersQuery = Query[ProjectUsersTable, ProjectUsersRecord, Seq]
+
+  object rawlsBillingProjectQuery extends TableQuery(new RawlsBillingProjectTable(_)) {
+
+    def save(billingProject: RawlsBillingProject): WriteAction[RawlsBillingProject] = {
+      val projectInsert = rawlsBillingProjectQuery insertOrUpdate marshalBillingProject(billingProject)
+      val userInserts = billingProject.users.toSeq.map {
+        projectUsersQuery insertOrUpdate marshalProjectUsers(_, billingProject)
+      }
+
+      projectInsert andThen findUsersByProjectName(billingProject.projectName.value).delete andThen DBIO.seq(userInserts: _*) map { _ => billingProject }
+    }
+
+    def load(rawlsProjectName: RawlsBillingProjectName): ReadAction[Option[RawlsBillingProject]] = {
+      val name = rawlsProjectName.value
+      findBillingProjectByName(name).result.flatMap {
+        case Seq() => DBIO.successful(None)
+        case Seq(projectRec) =>
+          for {
+            users <- findUsersByProjectName(name).result
+          } yield Option(unmarshalBillingProject(projectRec, users.toSet))
+        case _ => throw new RawlsException(s"Primary key violation: found multiple records for billing project '$name'")
+      }
+    }
+
+    def delete(billingProject: RawlsBillingProject): ReadWriteAction[Boolean] = {
+      val name = billingProject.projectName.value
+      val projectQuery = findBillingProjectByName(name)
+
+      projectQuery.result.flatMap {
+        case Seq() => DBIO.successful(false)
+        case Seq(projectRec) =>
+          findUsersByProjectName(name).delete andThen projectQuery.delete map { count => count > 0 }
+        case _ => throw new RawlsException(s"Primary key violation: found multiple records for billing project '$name'")
+      }
+    }
+
+    def addUserToProject(userRef: RawlsUserRef, billingProject: RawlsBillingProject): WriteAction[RawlsBillingProject] =
+      save(billingProject.copy(users = billingProject.users + userRef))
 
-  def deleteBillingProject(name: String) = rawlsBillingProjectByNameQuery(name).delete
+    def removeUserFromProject(userRef: RawlsUserRef, billingProject: RawlsBillingProject): WriteAction[RawlsBillingProject] =
+      save(billingProject.copy(users = billingProject.users - userRef))
+
```

Comment:
My intention was to match BillingDAO which returns the updated billing project.  But I see that this functionality is not actually used, so I'll update it to do the more sensible thing.